library/src/main/java/io/constructor/data/interceptor/TokenInterceptor.kt

```diff
@@ -1,19 +1,23 @@
 package io.constructor.data.interceptor
 
 import android.content.Context
+import io.constructor.BuildConfig
 import io.constructor.core.Constants
 import io.constructor.data.local.PreferencesHelper
 import okhttp3.Interceptor
 import okhttp3.Response
 
 
 
-class TokenInterceptor(val context: Context) : Interceptor {
+class TokenInterceptor(val context: Context, private val preferencesHelper: PreferencesHelper) : Interceptor {
 
     override fun intercept(chain: Interceptor.Chain): Response {
-        val token = PreferencesHelper(context).getToken()
         var request = chain.request()
-        val url = request.url().newBuilder().addQueryParameter(Constants.QueryConstants.AUTOCOMPLETE_KEY, token).build()
+        val url = request.url().newBuilder()
```

Comment:
If possible we should add session and identity to the token interceptor instead of applying on a per request basis.  All http requests should have `s`, `i`, `autocomplete_key`, `_dt`, and `c` parameters.